src/mapgen_valleys.cpp

```diff
@@ -521,35 +471,18 @@ float MapgenValleys::terrainLevelFromNoise(TerrainNoise *tn)
 	*tn->rivers = base;
 	if (river < 0.f) {
 		// Use the the function −sqrt(1−x²) which models a circle.
-		float depth = (river_depth * sqrt(1.f - pow((river / river_size + 1), 2)));
+		float depth = (river_depth_bed
+				* sqrt(1.f - pow((river / river_size_factor + 1), 2)));
```

Comment:
Are you sure that `pow((river / river_size_factor + 1), 2)` will never be greater than 1?